index.html

```diff
@@ -173,155 +181,311 @@
-            </dd>
-          </dl>
+          <ol>
+            <li link-for="VideoMonitorEvent">If <var>e</var> is
+            <a>videomonitor</a>, create a <a>VideoMonitorEvent</a>, initialize
+            it to have a given name <var>e</var>, to not bubble, to not be
+            cancelable.
+            </li>
+            <li link-for="VideoProcessorEvent">If <var>e</var> is
+            <a>videoprocess</a>, create a <a>VideoProcessorEvent</a>,
+            initialize it to have a given name <var>e</var>, to not bubble, to
+            not be cancelable.
+            </li>
+            <li link-for="VideoMonitorEvent">Initialize the <a>trackId</a>
+            attribute to the value of the <code>id</code> attribute of the <a>
+              video worker source</a>.
+            </li><!-- TODO -->
+            <li link-for="VideoMonitorEvent">Initialize the <a>playbackTime</a>
+            attribute to the value of the <code>currentTime</code> attribute of
+            the <a>MediaStream</a> that contains the <a>video worker
+            source</a>.
+            </li>
+            <li link-for="VideoMonitorEvent">Initialize the
+            <a>inputImageBitmap</a> attribute to the bitmap data of the
+            <a>video worker source</a>'s video frame at the current stream
+            position at <a>playbackTime</a>.
+            </li>
+            <li>
+              <!-- TODO: clarify the definition of targets -->
+              Let <var>associated video workers</var> be the list of
+              <a data-lt="video worker">video workers</a> that share the same
+              <a>video worker source</a>.
+            </li>
+            <li>If <var>e</var> is <a>videomonitor</a>, dispatch the newly
+            created <a>VideoMonitorEvent</a> object at each <var>target</var>.
+            </li>
+            <li link-for="VideoProcessorEvent">If <var>e</var> is
+            <a>videoprocess</a>, initialize the <a>outputImageBitmap</a>
+            attribute to null, and dispatch the newly created
+            <a>VideoProcessorEvent</a> object at each <var>target</var>.
+            </li>
+          </ol>
         </section>
-        <section>
-          <h2>
-            <code id = event-videomonitorevent>VideoMonitorEvent</code> interface
-          </h2>
+      </section>
+      <section>
+        <h2>
+          <code id="event-videomonitorevent">VideoMonitorEvent</code> interface
+        </h2>
+        <p>
+          <dfn>Video monitor events</dfn> use the <a>VideoMonitorEvent</a>
+          interface for their <a>videomonitor</a> events:
+        </p>
+        <pre class="idl">
+          [Constructor(DOMString type, optional VideoMonitorEventInit videoMonitorEventInitDict), Exposed=Worker]
+          interface VideoMonitorEvent : Event {
+              readonly    attribute DOMString   trackId;
+              readonly    attribute double      playbackTime;
+              readonly    attribute ImageBitmap? inputImageBitmap;
```

Comment:
The event attributes should be initialized to something. For symmetry with the output attribute I chose null, but could be something else too. What would you prefer for the default?

This is the usual prose that initializes the value:

>The inputImageBitmap attribute must return the value it was initialized to. When the object is created, this attribute must be initialized to null. It represents the ImageBitmap object whose bitmap data is provided by the video monitor source.

In "fire a video worker event named e", when we say in step 1 "create a VideoMonitorEvent", we initialize the attribute to null and in the step 5 that follows to the actual bitmap data.

Similarly to, for example in http://www.w3.org/TR/progress-events/#interface-progressevent

This means that if a web developer programmatically creates a new such event, these are the values the returned object is initialized to. This way the UA retains control over when the implementation will actually prime the input attribute with the real bitmap data (the UA decides when to fire these events). Is that how you intended this to behave? Let me know, and I'll update the PR accordingly.